storage/src/archive/immutable/storage.rs

```diff
@@ -117,13 +117,7 @@ impl<E: Storage + Metrics + Clock, K: Array, V: Codec> Archive<E, K, V> {
 
         // Get checkpoint
         let freezer_key = U64::new(FREEZER_PREFIX, 0);
-        let checkpoint = match metadata.get(&freezer_key) {
-            Some(freezer) => *freezer.freezer(),
-            None => {
-                metadata.put(freezer_key.clone(), Record::Freezer(Checkpoint::default()));
-                *metadata.get(&freezer_key).unwrap().freezer()
```

Comment:
When we load an empty table, we shouldn't initialize the checkpoint to be the default. That makes no sense.